docs/resources/ec_deployment.md

```diff
@@ -167,6 +168,15 @@ The optional `elasticsearch.remote_cluster` block can be set multiple times to r
 * `ref_id` (Optional) Remote elasticsearch `ref_id`, it is best left to the default value (Defaults to `main-elasticsearch`).
 * `ignore_unavailable` (Optional) If true, skip the cluster during search when disconnected (Defaults to `false`).
 
+##### Snapshot source
+
+The optional `elasticsearch.snapshot_source` block, which restores data from a snapshot of another deployment, supports the following arguments:
+
+* `source_cluster_id` (Required) ID of the Elasticsearch cluster that will be used as the source of the snapshot.
```

Comment:
I missed this in the previous review, is the ID the deployment ID? or the Elasticsearch Cluster ID? It seems that it's the Elasticsearch Cluster ID, not the deployment ID + `ref_id`. This can cause a lot of confusion for users if they're referencing the deployment ID instead of the ES cluster ID.

Maybe we can add a note that states that this ID is not to be confused with the deployment ID and that it's the Elasticsearch Resource ID.

Maybe it even makes sense to rename this field to `source_elasticsearch_resource_id`? it's extremely long, but very verbose in where the ID comes from.

Reply:
Yeah 🤔  I can see how that can be confusing